testing/e2e/api_version/client_api_current.go

```diff
@@ -88,6 +88,31 @@ func (tester *ClientAPITester) TestEnroll(apiKey string) (string, string) {
 	return enroll.Item.Id, enroll.Item.AccessApiKey
 }
 
+// TestEnrollUnavailable tests that the enroll endpoint is unavailable.
+func (tester *ClientAPITester) TestEnrollUnavailable(apiKey string) {
+	client, err := api.NewClientWithResponses(tester.endpoint, api.WithHTTPClient(tester.client), api.WithRequestEditorFn(func(ctx context.Context, req *http.Request) error {
+		req.Header.Set("Authorization", "ApiKey "+apiKey)
+		return nil
+	}))
+	tester.Require().NoError(err)
+
+	enrollResp, err := client.AgentEnrollWithResponse(tester.ctx,
+		&api.AgentEnrollParams{UserAgent: "elastic agent " + version.DefaultVersion},
+		api.AgentEnrollJSONRequestBody{
+			Metadata: api.EnrollMetadata{
+				Local: json.RawMessage(fmt.Sprintf(enrollMetadataTpl, version.DefaultVersion)),
+			},
+			Type: api.PERMANENT,
+		},
+	)
+	tester.Require().NoError(err)
+	tester.Assert().Condition(func() bool {
+		status := enrollResp.StatusCode()
+		tester.T().Logf("status code found: %v", status)
+		return status >= http.StatusInternalServerError
+	})
```

Comment:
```suggestion
	tester.Assert().GreaterOrEqual(enrollResp.StatusCode(), http.StatusInternalServerError)
```
(nit) I find the provided assertions easier to read, it's also what we're doing for all other checks